src/towncrier/test/test_settings.py

```diff
@@ -30,6 +30,61 @@ def test_base(self):
         self.assertEqual(config["filename"], "NEWS.rst")
         self.assertEqual(config["underlines"], ["=", "-", "~"])
 
+    def test_custom_types(self):
```

Comment:
can we these as 2 tests with separate docstring.

I am not sure that is the targeted difference between readme_format and alternative_format.

For the docstring, my suggetion is something like this

```
Custom fragment categories can be defined inside the toml config file using implicit file extension.
The file extension is the same as directory name.
```


```
Custom fragment categories can be defined inside the toml config file using custom file extension.
The file extension is define by the section name.
```

Does these docstrings make sense?
Is this what it is tested?

You can refactor the shared code in a helper function...but I think that it helps to have 2 separate test functions with 2 docstrings as they are covering 2 separate uses cases.

Thanks!

Reply:
Done.